addons/account/models/account_move_line.py

```diff
@@ -804,7 +804,10 @@ def _compute_product_uom_id(self):
     @api.depends('display_type')
     def _compute_quantity(self):
         for line in self:
-            line.quantity = 1 if line.display_type == 'product' else False
+            if line.display_type == 'product':
+                line.quantity = 1 if not line.quantity else line.quantity
```

Comment:
umm, it should be reset to 1 anyways since  an aml with quantity 0 doesn't make sense no ?

Reply:
I'd go for JB's suggestion, I actually stopped on that one too but considered it would be too corner-casey
At least it ensures the totals stay the same, and errors raised when needed (if any)